raiden/network/transport/matrix/utils.py

```diff
@@ -64,6 +64,35 @@ class AddressReachability(Enum):
 }
 
 
+class DisplayNameCache:
+    def __init__(self) -> None:
+        self._userid_to_displayname: Dict[str, str] = dict()
+
+    def warm_users(self, users: List[User]) -> None:
+        for user in users:
+            user_id = user.user_id
+            cached_displayname = self._userid_to_displayname.get(user_id)
+
+            if cached_displayname is None:
+                # The cache is cold, query and warm it.
+                if not user.displayname:
+                    user.get_display_name()
```

Comment:
https://github.com/matrix-org/synapse/blob/782dd72037cf71fb3f9e4922b07c56df2f59de75/synapse/handlers/profile.py#L137-L140 need to be handled here